src/Microsoft.AspNetCore.SystemWebAdapters/Adapters/ProxyHeaderModule.Framework.cs

```diff
@@ -59,11 +102,11 @@ private static void UseHeaders(HttpRequest request)
             {
                 request.ServerVariables.Set("SERVER_PORT", value.Port);
             }
-        }
 
-        if (request.Headers["x-forwarded-proto"] is { } proto)
-        {
-            request.ServerVariables.Set("SERVER_PROTOCOL", proto);
+            if (request.Headers["x-forwarded-proto"] is { } proto)
```

Comment:
do we not want to set the proto if the host is not available?

Reply:
It seemed like a weird scenario to set a different proto without changing host or port. Is there any scenario where we would forward from http to https or vice-versa with the same host and port?

Reply:
Reverted since I'm no longer touching this class for any other reason and it doesn't seem important enough to change on its own.